### ISSUE TITLE
Better Metrics: One Metric, multiple history entries

### ISSUE THREAD

**Issue by Drenderer** [Drenderer/klax#108]
## Problem description
When a loss function consists of multiple terms, e.g.,
```python
@klax.loss
def my_loss(model, batch, run_state):
    ... # computation
    return loss_a + loss_b + 5*loss_c
```
I often want to record the individual terms `loss_a`, `loss_b` and `loss_c` in the history. This is easy, if their computation can be separated completely:
```python
class MyLoss(klax.Loss):
    def loss_a(self, model, batch, run_state):
        ...
        return loss_a

    def loss_b(self, model, batch, run_state):
        ...
        return loss_b

    def loss_c(self, model, batch, run_state):
        ...
        return loss_c

    def value(self, model, batch, run_state):
        return (
            self.loss_a(model, batch, run_state)
            + self.loss_b(model, batch, run_state)
            + 5 * self.loss_c(model, batch, run_state)
        )
```
Then one can define Metrics for each loss term:
```python
my_loss = MyLoss()
metric_a= klax.BatchMetric(
    "loss_a",
    lambda model, batch, run_state: my_loss.loss_a(klax.unwrap(model), batch, run_state)
    ...
)
metric_b = ...
...
```
However, if the computation of the individual terms cannot be entangled (e.g. both `loss_a` and `loss_b` rely on the same intermediate quantity), then this would mean recomputing the intermediates for each loss term.
To avoid this, one can define the loss like this:
```python
class MyLoss(klax.Loss):
    def _loss_terms(self, model, batch, run_state):
        ... # computation
        return loss_a, loss_b, loss_c

    def loss_terms(self, model, batch, run_state):
        _model = klax.unwrap(model)
        return self._loss_terms(_model, batch, run_state)

    def value(self, model, batch, run_state):
        loss_a, loss_b, loss_c = self._loss_terms(model, batch, run_state)
        return loss_a + loss_b + 5*loss_c
```
And use a Metric like this:
```python 
my_loss = MyLoss()
my_metric = klax.BatchMetric(
    "loss-terms",
    my_loss.loss_terms,
    ...
)
```
Now the computation is more efficient, but the issue is that in the history, the three terms are not separated. While plotting the history still works, each line for the different loss terms will have the same label (`"loss-terms"`), making it difficult to keep them apart. And extracting the data from the history to reformat it is quite cumbersome.

## Proposed solution
It would be beneficial, if there was a `BatchMultiMetric` (The metrics should really be called `BatchedMultiMetric`) that accepts a list of names:
```python
my_metric = klax.BatchMultiMetric(
    ("loss_a", "loss_b", "loss_c"),
    my_loss.loss_terms,
    ...
)
```
The `MetricLogger` then would need to write multiple entries into the `History` - one for each name in the BatchMultiMetric. This would essentially allow one function to compute multiple metrics at the same time.

It would also be possible to bake this behavior into the existing `BatchMetric` - although that's probably less desirable.
